### DISCUSSION TITLE
Move code to another project?

### DISCUSSION THREAD

**Discussion by bbarker** [unisonweb/unison#5488] [Q&A]
Is there a recommended way to move terms or namespaces to another project?

I suppose one way would be to make the "old" project a dependency of the new project, move things, then delete the dependency, but not sure if this is ideal.

**Comment by mitchellwrosen** [comment]
I'm not sure we have a great way of accomplishing this at the moment.

There is only a self-contained-ness constraint at the project level (that is, a term `foo.bar.myterm` within a project can have dependencies outside of the sub-namespace `foo.bar`, so it's a bit unclear how we want to handle a user's attempt at moving a term into a project in which there are not names for all of its (transitive) dependencies.

That said... we do have a syntax for namespaces within a project: `project:path`. It's not implemented in every command, but it is implemented in `fork`. So you can `fork project1:path1 project2:path2`, it just might leave you with unnamed references (run `todo` in `project2` to confirm).

**Comment by bbarker** [comment]
Just as a note for future reference, I ended up using this exact syntax:

```
fork @bbarker/bbblog/main:utils.html html-util/main:utils.html
```

Otherwise i'd get some errors:

```
@bbarker/bbblog/main> fork bbblog:utils.html html-util:utils.html

  ⚠️

  Sorry, I wasn’t sure how to process your request:

    <none>:1:7:
      |
    1 | bbblog:utils.html
      |       ^
    unexpected ':'
    expecting '/' or end of input


  You can run `help fork` for more information on using `fork`.
 ```
 
 
 ```
 @bbarker/bbblog/main> fork bbblog/main:utils.html html-util/main:ut

  bbblog/main does not exist.
  ```
